lib/rules/no-loss-of-precision.js

```diff
@@ -44,9 +52,10 @@ module.exports = {
          */
         function isBaseTen(node) {
             const prefixes = ["0x", "0X", "0b", "0B", "0o", "0O"];
+            const raw = getRaw(node);
 
-            return prefixes.every(prefix => !node.raw.startsWith(prefix)) &&
-            !/^0[0-7]+$/u.test(node.raw);
+            return prefixes.every(prefix => !raw.startsWith(prefix)) &&
+            !/^0[0-7]+$/u.test(raw);
         }
```

Comment:
Maybe this function doesn't have to be changed, so we could avoid one extra `getRaw` call?

Numeric separators shouldn't affect the `node.raw.startsWith(prefix)` check.

As for the `!/^0[0-7]+$/u.test` condition, legacy octal literals can't have numeric separators.

Reply:
Thanks for the points,  having a separator there would be thrown by the parser as well.